spec/inputs/elasticsearch_spec.rb

```diff
@@ -92,9 +92,11 @@
 
         before do
           allow(Elasticsearch::Client).to receive(:new).and_return(es_client)
-          allow(es_client).to receive(:info).and_raise(
-            Elasticsearch::Transport::Transport::Errors::BadRequest.new
-          )
+          begin
+            allow(es_client).to receive(:info).and_raise(Elasticsearch::Transport::Transport::Errors::BadRequest.new)
+          rescue NameError # NameError: uninitialized constant Elasticsearch::Transport
+            allow(es_client).to receive(:info).and_raise(Elastic::Transport::Transport::Errors::BadRequest.new)
+          end
```

Comment:
I think that being more explicit on the intention would help to understand better:
```ruby

def elastic_ruby_client8_available?
  Elasticsearch::Transport
  return false
rescue NameError # NameError: uninitialized constant Elasticsearch::Transport if Elastic Ruby client is not available
  return true
end


before do
  ...
  unless elastic_ruby_client8_available?
    allow(es_client).to receive(:info).and_raise(Elasticsearch::Transport::Transport::Errors::BadRequest.new)
  else
    allow(es_client).to receive(:info).and_raise(Elastic::Transport::Transport::Errors::BadRequest.new)
  end
  ...
```

I know it's more verbose but maybe more explicit for the reader.